.github/workflows/main.yml

```diff
@@ -13,27 +13,130 @@ on:
 
 jobs:
   build:
-    runs-on: ${{ matrix.os }}
-
+    name: ${{ matrix.config.name }}
+    runs-on: ${{ matrix.config.os }}
+    defaults:
+      run:
+        shell: ${{ matrix.config.shell }}
     strategy:
+      fail-fast: false
       matrix:
-        compiler: [clang, gcc]
-        os: [ubuntu-latest]
+        config:
+          - name: Linux GCC
+            os: ubuntu-latest
+            compiler: gcc
```

Comment:
I'm just wondering if we lose anything here by not also testing Ubuntu with `clang`, unless it's considered to be fully covered by macOS.

Reply:
I think macOS is enough.

Reply:
> I think macOS is enough.

dito